### PR TITLE
[feature](FE): add new property to control whether use light schema change or not

### PR DESCRIPTION
**Pull request by ByteYue**
# Proposed changes

Issue Number: 

## Problem Summary:

Add one new property `light_schema_change` to control whether use the light schema change optimization. When it's set true, Doris would try light schema change when alter schema change job. It's turned  by default, which means user should use `light_schema_change = true` if they wan to use the optimization.

## Checklist(Required)

1. Does it affect the original behavior: Yes
2. Has unit tests been added: No
3. Has document been added or modified: Yes
4. Does it need to update dependencies: Yes
5. Are there any changes that cannot be rolled back: No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java [hunk 1]
```diff
@@ -451,8 +453,27 @@ public static long analyzeTimeout(Map<String, String> properties, long defaultTi
         return timeout;
     }
 
+    public static Boolean analyzeUseLightSchemaChange(Map<String, String> properties) throws AnalysisException {
+        String useSchemaChange = "";
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java:457]: Use boolean instead of String directly？

**Review by dataroaring — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java (hunk 1 above)
Comment (ByteYue) [reply fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java]: I adapt this function using the similar logic as analyzeUniqueKeyMergeOnWrite below.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalDataSource.java [hunk 2]
```diff
@@ -1718,6 +1713,26 @@ private void createOlapTable(Database db, CreateTableStmt stmt) throws UserExcep
         // this should be done before create partition.
         Map<String, String> properties = stmt.getProperties();
 
+        // get use light schema change
+        Boolean useLightSchemaChange = true;
+        try {
+            useLightSchemaChange = PropertyAnalyzer.analyzeUseLightSchemaChange(properties);
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalDataSource.java:1719]: Does useLightSchemaChange only control whether a column has unique id? What happens when alter table?

Comment (ByteYue) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/InternalDataSource.java:1719]: Yes, this new property only affects whether the column would be assigned a unique id or not.
Alter table job will try to use the light schema change optimization (like add new value column) if the columns of the tables are assigned unique id.

**Comment by Lchangliang** [comment]
LGTM

**Review by stalary — DISMISSED** [review]
LGTM

**Comment by jacktengg** [comment]
Please fix FE code style check error.

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowAlterStmt.java [hunk 3]
```diff
@@ -104,12 +104,13 @@ private void getPredicateValue(Expr subExpr) throws AnalysisException {
         if (!(subExpr instanceof BinaryPredicate)) {
             throw new AnalysisException("The operator =|>=|<=|>|<|!= are supported.");
         }
+        LOG.info("enter getPredicateValue");
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ShowAlterStmt.java:107]: remove the log

File: fe/fe-core/src/main/java/org/apache/doris/common/util/TimeUtils.java [hunk 4]
```diff
@@ -149,13 +152,22 @@ public static String longToTimeString(long timeStamp, SimpleDateFormat dateForma
         return dateFormat.format(new Date(timeStamp));
     }
 
-    public static String longToTimeString(long timeStamp) {
-        SimpleDateFormat datetimeFormatTimeZone = datetimeFormatThreadLocal.get();
+    public static String longToTimeStringProc(long timeStamp, SimpleDateFormat datetimeFormatTimeZone) {
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/TimeUtils.java:155]: longToTimeStringProc, the name is a little strange. Maybe we can rename it to longToTimeStringWithFormat?

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalDataSource.java [hunk 5]
```diff
@@ -1731,6 +1726,26 @@ private void createOlapTable(Database db, CreateTableStmt stmt) throws UserExcep
         // this should be done before create partition.
         Map<String, String> properties = stmt.getProperties();
 
+        // get use light schema change
+        Boolean useLightSchemaChange = true;
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalDataSource.java:1730]: default should be false？

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalDataSource.java [hunk 6]
```diff
@@ -1731,6 +1726,26 @@ private void createOlapTable(Database db, CreateTableStmt stmt) throws UserExcep
...
+        try {
+            useLightSchemaChange = PropertyAnalyzer.analyzeUseLightSchemaChange(properties);
+        } catch (AnalysisException e) {
+            throw new DdlException(e.getMessage());
+        }
+        // use light schema change optimization
+        olapTable.setUseLightSchemaChange(useLightSchemaChange);
+        if (useLightSchemaChange) {
+            LOG.info("table: {} uses light schema change", olapTable.getName());
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalDataSource.java:1739]: LOG.debug is enough.

**Review by dataroaring — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalDataSource.java (hunk 5 above)
Comment (ByteYue) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/InternalDataSource.java:1730]: Actually it wouldn't have any effect cause useLightSchemaChange would be assigned value by PropertyAnalyzer.analyzeUseLightSchemaChange which is set to false by default. But it's confusing indeed. Will change it soon.

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**